File: Week1/DiagonalDifference/src/main/java/MyCode.java
package main.java;

import java.util.List;

public class MyCode {
    public static int diagonalDifference(List<List<Integer>> arr) {

        int firstSum = 0;
        int secondSum = 0;
        for (int i = 0; i < arr.size();i++){
            firstSum += arr.get(i).get(i);
            secondSum += arr.get((arr.size()-1)-i).get(i);
        }
        return Math.abs(firstSum-secondSum);
    }
}

File: Week1/DiagonalDifference/src/main/kotlin/MyCode.kt
package main.kotlin

fun diagonalDifference(arr: Array<Array<Int>>): Int{
    var firstSum = 0
    var secondSum = 0
    for (row in arr.indices){
        firstSum += arr[row][row]

    }
    var counterAux = 0
    for (row in (0..arr.lastIndex).reversed()){
        secondSum += arr[row][counterAux]
        counterAux++
    }
    return Math.abs((secondSum - firstSum))
}
fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val arr = Array<Array<Int>>(n, { Array<Int>(n, { 0 }) })

    for (i in 0 until n) {
        arr[i] = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()
    }

    val result = diagonalDifference(arr)

    println(result)
}